packages/lit-dev-content/site/tutorials/content/carousel/07.md

```diff
@@ -0,0 +1,50 @@
+Now you're ready to position the elements.
+
+Update the top of the `render` method to look like this:
+
+```ts
+render() {
+  const p = this.selectedInternal;
+  const s = (this.selectedInternal =
+    this.hasValidSelected() ? this.selected : this.selectedInternal);
+  const shouldMove = this.hasUpdated && s !== p;
+  const atStart = p === 0;
+  const toStart = s === 0;
+  const atEnd = p === this.maxSelected;
+  const toEnd = s == this.maxSelected;
+  const shouldAdvance = shouldMove &&
+    (atEnd ? toStart : atStart ? !toEnd : s > p);
+  const delta = (shouldMove ? Number(shouldAdvance) || -1 : 0) * 100;
+  this.left -= delta;
+  const animateLeft = `${this.left}%`;
+  const selectedLeft = `${-this.left}%`;
+  const previousLeft = `${-this.left - delta}%`;
+```
+
+There's a bit of code here to review. It leverages the previously added code
+to help position the elements.
+
+First, `shouldMove` is computed based on the `selectedInternal`
+tracking property changing and using ReactiveElement's `hasUpdated` property
+to avoid animating the first render.
+
+Then, `shouldAdvance` is computed so that the position advances if the selected
+item increases, with the positioning reversed when wrapping to the start or end.
+
+The move `delta` is calculated based on `shouldMove` and `shouldAdvance`, and
+using the `delta`, the `left` property is set. It stores a `%` value indicating
+where the slot "container" element should be positioned. It's stored as state
+so the element re-renders correctly if it updates and the `selected` property
+has not changed.
```

Comment:
> left is stored as state

was quite confusing to me. If you don't mind explaining what you mean here arthur and I can tackle it next week

Reply:
Will take a stab at cleaning up: it's stored in a property rather than computed ephemerally so that if nothing changes, the left can just re-apply where it is. Could also conditionally use noChanged + the styleMaps, but this was simpler.